src/routes/popup/arns/setPrimaryName/ArNSConfirmSetPrimaryNameView.tsx

```diff
@@ -0,0 +1,128 @@
+import { ButtonV2 } from "@arconnect/components";
+import { Text } from "@arconnect/components-rebrand";
+import { useMemo, useState } from "react";
+import { Flex } from "~components/common/Flex";
+import HeadV2 from "~components/popup/HeadV2";
+import { ARNS_QUERY_CLIENT, setPrimaryName, useArioBalance, usePrimaryNameCostDetails, useTicker } from "~lib/arns";
+import { NAME_SERVICE_QUERY_CLIENT } from "~lib/nameservice";
+import { useActiveWallet } from "~wallets/hooks";
+import type { CommonRouteProps } from "~wallets/router/router.types";
+import { useLocation } from "~wallets/router/router.utils";
+import { CardContainer } from "../CardContainer";
+import TransactionStatusModal from "../TransactionStatusModal";
+import { sleep } from "~utils/promises/sleep";
+import { formatArio } from "../utils";
+import { mARIOToken } from "@ar.io/sdk/web";
+import { ProfileName } from "~components/ProfileName";
+
+export interface ArNSConfirmSetPrimaryNameViewParams {
+  name: string;
+}
+export type ArNSConfirmSetPrimaryNameViewProps = CommonRouteProps<ArNSConfirmSetPrimaryNameViewParams>;
+
+export const ArNSConfirmSetPrimaryNameView = ({ params: { name } }: ArNSConfirmSetPrimaryNameViewProps) => {
+  const [processingTransaction, setProcessingTransaction] = useState<boolean>(false);
+
+  const [transactionState, setTransactionState] = useState<string | undefined>();
+
+  const { data: costDetails } = usePrimaryNameCostDetails({ name });
+  const { data: ticker } = useTicker();
+
+  const costDetailsArio = useMemo(() => {
+    return costDetails ? new mARIOToken(costDetails.tokenCost).toARIO().valueOf() : undefined;
+  }, [costDetails]);
+  const formattedCost = useMemo(() => {
+    return costDetails ? formatArio(costDetailsArio) : undefined;
+  }, [costDetailsArio]);
+
+  const arioBalance = useArioBalance();
```

Comment:
You could use hook [useTokenBalance](https://github.com/ar-io/ArConnect/blob/ebd0e85a69eff031ac80edce49e0b13a30cc0e67/src/tokens/hooks/index.ts#L85) for fetching token balance instead of `useArioBalance` as it uses tanstack caching and don't forget to invalidate balance after some purchases after using `useTokenBalance`.

Reply:
Updated in https://github.com/wanderwallet/Wander/pull/1001/commits/f91df79f6bea1b56de366c079cee56ba5fa50580 to use useTokenBalance within useArioBalance(). Ended up keeping useArioBalance since it handles parsing to float, and didn't want to repeat code in the places that currently use useArioBalance. 

Also updated the screens that do transaction processing to invalidate the balance query for updating.